src/main/java/org/folio/search/controller/ApiExceptionHandler.java

```diff
@@ -0,0 +1,148 @@
+package org.folio.search.controller;
+
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+
+import java.util.Optional;
+import javax.validation.ConstraintViolationException;
+import lombok.extern.log4j.Log4j2;
+import org.apache.commons.lang3.StringUtils;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.index.Index;
+import org.folio.search.domain.dto.Error;
+import org.folio.search.domain.dto.ErrorResponse;
+import org.folio.search.domain.dto.Parameter;
+import org.folio.search.exception.SearchOperationException;
+import org.folio.search.exception.SearchServiceException;
+import org.folio.search.model.types.ErrorCode;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@Log4j2
+@RestControllerAdvice
+public class ApiExceptionHandler {
+
+  /**
+   * Catches and handles all {@link SearchServiceException} objects during code execution.
+   *
+   * @param exception {@link SearchServiceException} to process
+   * @return {@link ResponseEntity} with {@link ErrorResponse} body
+   */
+  @ExceptionHandler(SearchServiceException.class)
+  public ResponseEntity<ErrorResponse> handleSearchServiceException(SearchServiceException exception) {
+    log.debug("Handling exception [type: SearchServiceException]", exception);
```

Comment:
Should it be warn or at least info?

Reply:
This exception can occur frequently - for example when user provides invalid CQL search query, I don't see the reason to log every validation error at info or warn level